mount-partitions.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# Mounts boot, state, & data partitions from balenaOS.
+# The container must be privileged for this to function correctly.
+
+# Get the current boot block device in case there are duplicate partition labels
+# for `(balena|resin)-(boot|state|data)` found.
+# Returns sdaX, mmcblk0pX, nvme0n1pX, etc.
+current_boot_partition=$(fdisk -l | grep '*  ' | awk '/^\/dev/{sub("/dev/","",$1); print $1}')
+# Returns /dev/sda, /dev/mmcblk0, /dev/nvme0n1, etc. This is the 
+current_boot_block_device="/dev/$(basename "$(readlink -f "/sys/class/block/${current_boot_partition}"/..)")"
```

Comment:
I made this change in the latest commit because before, we were running `fdisk -l | grep '*  ' | cut -d' ' -f1 2>&1`, which would return the boot partition device, `/dev/sda1` / `/dev/nvme0n1p1` / `/dev/mmcblk0p1`, however, to see whether other partitions belong to the current block device, we need to remove the partition identifier from those devices. That identifier differs between sda, nvme, and mmcblk - it's either simply a number or `pN`. I've tested this on an `sda` block device and an `nvme0n1` block device and it works on both, so should be the more flexible way to get the current boot block device.

Reply:
I'm not entirely sure we should rely on the folder structure across kernels/os versions. I searched for a standard way to do this, and it looks like `lsblk` is the tool you might want to use  https://unix.stackexchange.com/a/308726

```
root@b308f87:~# lsblk -no pkname /dev/mmcblk0p1
mmcblk0
```

Reply:
Changed it to use lsblk (and install in Dockerfile) here: 
https://github.com/balena-os/balena-supervisor/pull/2120/files#diff-8a96bc9af661488f86b92785e9382315f61c548974c4bcce92c1a1deb9a129f8R8-R15